cmd/minishift/cmd/config/config.go

```diff
@@ -100,6 +100,14 @@ var settings []Setting = []Setting{
 		callbacks:   []setFn{RequiresRestartMsg},
 	},
 	{
+		name: "hostfolders-automount",
```

Comment:
> Host folders on instance level seemed on the previous PR the preferred way of defining them.

I guess you are referring to https://github.com/minishift/minishift/pull/526#discussion_r106385226. I forgot to answer to respond on this one. I think there is a misunderstanding here.

> TL;dr: mountpath is defined as default /mnt/sda1, but allow for override on global level to be something else. On instancelevel each HostFolder has a mountpoint that overrides the auto generated mountpoints

Sure, but it means I have to re-define the whole mount configuration over and over again. Leaving the instance config aside for a second. The whole point of 'config set' is that one does not have to repeat the same parameters over and over again. Once defined I can just do 'minishift start && minishift delete', repeat. At the moment this is the main use case, especially since we don't support the multiple instance approach yet. 

Once we are going to implement this full out, we need to make some decisions anyways. For example do we have completely isolated instances ( in which case things like vm-driver, cpus, etc need to be instance specific as well) or is there a part of the config which applies to all instances created. There are several things up in the air. Really the introduction if the instance specific minishift.json was in this light maybe even to early. However, the oc path felt definitely like something which is instance specific and transparent to the user. 

So let's take another angle on this. Leaving aside technicalities on which JSON file is used and leaving aside that we are going to support multiple instances. As it stands now, as a user what would you expect? Also taken into account what 'config set' does. Would you expect you add the hostfolder and then you can re-create VMs without having to do it over and over or would you expect that these settings need to be re-entered each single time? For me this is the core question. 

@praveenkumar, @LalatenduMohanty - you have not said anything to this yet. What is your take on this one?